app/src/main/java/com/github/openwebnet/repository/impl/EnvironmentRepositoryImpl.java

```diff
@@ -102,4 +106,19 @@ public EnvironmentRepositoryImpl() {
         });
     }
 
+    @Override
+    public Observable<Void> delete(Integer id) {
+        return Observable.create(subscriber -> {
+            try {
```

Comment:
You can try but Realm is a bit fussy, currently is still not possible to extend anything else than RealmObject or to override methods like toString() or equals()